app/scripts/components/common/map/style-generators/raster-timeseries.tsx

```diff
@@ -613,5 +501,14 @@ export function RasterTimeseries(props: RasterTimeseriesProps) {
   );
   useFitBbox(!!isPositionSet, bounds, layerBounds);
 
-  return null;
+  return (
+    points && (
+      <PointsLayer
+        id={id}
+        points={points}
+        zoomExtent={zoomExtent}
+        onPointsClick={onPointsClick}
+      />
+    )
+  );
```

Comment:
Can you explain why the points are returned from the raster-timeseries layer?

I may be missing some context, but I think the decision of whether to display points or raster based on zoom level would make more sense at a higher architectural level, before rendering either.
It seems confusing to have one layer returned from another, and leaves the responsibility unclear. A cleaner approach would be to have a decision layer that selects the appropriate visualization method (points or raster) based on zoom level, then routes to either a dedicated points layer or raster layer.

Reply:
The points are there to indicate where the raster datasets are. When a user is on a lower zoom than the dataset's minimum zoom, the points will show up. From this context that points data is derived from raster timeseries data, I think it makes sense that points are accompanied by raster paint layer, and raster-timeseries is that decision layer to render both of those layers. It would make sense to make the minZoom/maxZoom more explicit as you suggested, and I think it would be on JSX level change (in pseudo code :  `points && map.minZoom < ... && <Points ... ` than having one more layer on top of raster timeseries. Maybe I am looking at this code too long that I am missing something? In any case, let's make it better little by little 🔨